app/javascript/textarea-markdown-linkify.js

```diff
@@ -0,0 +1,37 @@
+import MarkdownIt from 'markdown-it'
+
+export default class {
+  linkify(selector) {
+    const textareas = document.querySelectorAll(selector)
+    if (textareas.length === 0) {
+      return null
+    }
+
+    Array.from(textareas).forEach((textarea) => {
+      textarea.addEventListener('paste', async (event) => {
+        event.preventDefault()
+        const { selectionStart, selectionEnd } = textarea
+        const selectedText = textarea.value.slice(selectionStart, selectionEnd)
+        const escapedSelectedText = selectedText.replace(/[[\]]/g, '\\$&')
+        // headless chromeではevent.clipboardData.getData('text')は空文字を返すため、代わりにnavigator.clipboard.readText()を使用
+        // https://github.com/fjordllc/bootcamp/pull/6747#discussion_r1325362833
+        const pasteText =
+          event.clipboardData.getData('text') ||
+          (await navigator.clipboard.readText())
+        if (selectedText && this._isURL(pasteText)) {
+          const markdownLink = `[${escapedSelectedText}](${pasteText})`
+          document.execCommand('insertText', false, markdownLink)
+        } else {
+          document.execCommand('insertText', false, pasteText)
+        }
```

Comment:
```suggestion
        const textToInsert = (selectedText && this._isURL(pasteText)) ? `[${escapedSelectedText}](${pasteText})` : pasteText
        document.execCommand('insertText', false, textToInsert)
```

より短く、かつDRYにするならこんなアプローチもあります。

Reply:
修正しました。[700c94c](https://github.com/fjordllc/bootcamp/pull/6747/commits/700c94cf2c9ec0ec3ceed8a737d7beb9b5e83cdd)